### PR TITLE
Ci/test x plat

### PR DESCRIPTION
**Pull request by YOU54F**
This PR address lint errors introduced by #347 

Also looks to test the Pact Go V2 project on windows and macos in GHA, and on arm64 hardware for linux/macos.

This also recreates the issue noted in #345 and goes some way towards #297 

I need to pull out the changes from https://github.com/pact-foundation/pact-go/pull/351/commits/9738a2b8dd2a1774985a89357faa123da1b998d1 but just thought I would stick this as a draft, in case people are wondering the golang-lintci is failing

### CODE REVIEW

**Comment by YOU54F** [comment]
ps. failing tests in cirrus ci (and cirrus cli locally) but everything passing on my local mac and also passing on GHA so 🤷🏾‍♂️

File: .cirrus.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+test: &TEST
+  test_script: make test
+
+linux_arm64_task:
+  env:
+    SKIP_PLUGIN_AVRO: "true"
```
[the remark this answers is not in the document]
Comment (YOU54F) [reply .cirrus.yml:6]: yeah avro was failing, needs java plus other yaks.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -57,3 +57,25 @@ jobs:
       with:
         github-token: ${{ secrets.GITHUB_TOKEN }}
         parallel-finished: true
+
+  test-other-os:
```
[the remark this answers is not in the document]
Comment (YOU54F) [reply .github/workflows/test.yml:61]: Thought 

1) try and get build and unit tests working first x plat
2) then move onto examples
    - consider examples as 2nd class compared to main code that is currently untested in CI x-plat

 not sure how yaks we would need to end up shaving to get this working, so thought I would just create a seperate job to check whats needed for these non linux os's, and then could roll everything back together intro one job

[the remark this answers is not in the document]
Comment (mefellows) [reply .github/workflows/test.yml:61]: Cool, I thought so. It's better than where we were, but just wanted to clarify.

**Comment by YOU54F** [comment]
round 2 in #416

**Pull request closed without merging**